mlir/include/mlir/Dialect/SCF/IR/SCFOps.td

```diff
@@ -153,9 +153,10 @@ def ForOp : SCF_Op<"for",
     operation defines an SSA value for its induction variable. It has one
     region capturing the loop body. The induction variable is represented as an
     argument of this region. This SSA value is a signless integer or index.
-    The step is a value of same type but required to be positive. The lower and
-    upper bounds specify a half-open range: the range includes the lower bound
-    but does not include the upper bound.
+    The step is a value of same type but required to be positive, the lower and
+    upper bounds can be also negative or zero. The lower and upper bounds specify
+    a half-open range: the range includes the lower bound but does not include
+    the upper bound.
```

Comment:
Thanks for updating this. I would explicitly mention that an iteration is executed iff the signed comparison of induction variable value is less than the upper bound. WDYT?
@joker-eph might have some better suggestion on wording.

Reply:
Seems like a good addition!

Reply:
@ThomasRaoux @joker-eph sorry for late respond, was AFK for some time.
I updated the description as mentioned above, can you please review?